src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -344,22 +348,18 @@ public void Login()
         public bool OpenRepository()
         {
             var old = Repositories.FirstOrDefault(x => x.Equals(Holder.ActiveRepo));
-            // open the solution selection dialog when the user wants to switch to a different repo
-            // since there's no other way of changing the source control context in VS
             if (!Equals(SelectedRepository, old))
             {
-                if (ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
+                var opened = vsServices.TryOpenRepository(SelectedRepository.LocalPath);
```

Comment:
If this one fails, fallback to the previous codepath that prompts for a solution, that way there's a clear user interaction that the user can cancel out of, and we won't have to specifically explain why our initial attempt failed

```
(ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
```